agent/engine/task_manager.go

```diff
@@ -690,3 +699,42 @@ func (mtask *managedTask) waitForStopReported() bool {
 	}
 	return taskStopped
 }
+
+// sendExecutionStoppedAtTimestamp start the timer for sending execution stopped at timestamp
```

Comment:
What exactly is this timer used for?

Reply:
We want this timestamp to be sent along with the task STOPPED event to save 
 one extra SubmitTaskSTateChange call, but if the task failed to stop within `executionStoppedAtSubmitDelay` this timer will be triggered to send the timestamp.

Reply:
As per offline conversation, you shouldn't need to do this at all. The ticker in `eventhandler.TaskHandler` should take care of batching and sending all unsent events.